src/nvm_x722.cpp

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include "nvm_x722.hpp"
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <system_error>
+
+NvmX722::NvmX722(const std::filesystem::path& file)
+{
+    int fd = -1;
+    data = MAP_FAILED;
+
+    try
+    {
+        // open file
+        fd = open(file.c_str(), O_RDWR);
+        if (fd == -1)
+        {
+            throw std::system_error(errno, std::generic_category());
```

Comment:
Exception in the constructor? I believe it's a bad idea.
Besides, we already have a wrapper around the `mmap` (see `MappedMem` in `src/signature.cpp`). Probably it would be better to move it into standalone header/cpp file and extends for using in read-write mode.

Reply:
i'm sure you are confusing constructor and destructor.
Stroustrup tells us that we "should throw an exception from a constructor": https://www.stroustrup.com/bs_faq2.html#ctor-exceptions

Reply:
I still believe that the pure constructor and static method returning the object or throwing an exception is better and more readable.

Reply:
Sorry, you haven't convinced me =)
I believe we could use C++ exception in constructors.
AFAIK, the "two phase constructors" idiom is mostly used in the "C with classes" or with systems that have limited exception support (such as SymbianOS).